dashboard-server/queries/responses/response.queries.js

```diff
@@ -13,7 +13,10 @@ async function all() {
        SELECT DISTINCT ON (1) userid, timestamp AS last_timestamp, response AS last_response, surveyid
        FROM   responses
        ORDER  BY 1,2 DESC
-       ) l USING (userid)`;
+       ) l
+    USING (userid)
+    ORDER BY first_timestamp DESC
```

Comment:
Interestingly, if not sorted this query returns results in different orders unpredictably.

Reply:
Yep, it's a distributed database, the order of the partitions coming back isn't guaranteed!